setup.sh

```diff
@@ -59,109 +83,157 @@ function clean_legacy()
   local trash=$(mktemp -d)
   local completely_remove=$1
 
-  local toDelete="$APPLICATIONNAME"
-  echo_n_run eval "sed -i '/$toDelete/d' $HOME/.bashrc"
-  if [[ $completely_remove =~ "-d" ]]; then
-    echo_n_run mv "$INSTALLTO" "$trash"
-    return 0
+  mkdir -p "$trash/bin" "$trash/share" "$trash/lib" "$trash/cache/"
+  echo_n_run mv "$binpath" "$trash/bin/"
+  echo_n_run mv "$libdir" "$trash/lib/"
+  echo_n_run mv "$sharedir" "$trash/share/"
+  echo_n_run mv "$cachedir" "$trash/cache/"
+
+  if [[ "$system_wide" == true ]]; then
+    echo_n_run mv "$manpath" "$trash/man"
+    echo_n_run mv "$bashcompletionpath" "$trash/bash-completion"
+    echo_n_run mv "$fishcompletionpath" "$trash/fish-completion"
   fi
 
-  # Remove files
-  if [ -d "$INSTALLTO" ]; then
-    # If we have configs, we should keep it
-    if [ -d "$INSTALLTO/$CONFIGS_PATH" ]; then
-        for content in "$INSTALLTO"/*; do
-          if [[ $content =~ "configs" ]]; then
-            continue
-          fi
-          echo_n_run mv "$content" "$trash"
-        done
-    else
-      echo_n_run mv "$INSTALLTO" "$trash"
+  if [[ $completely_remove =~ "-d" ]]; then
+    mkdir -p "$trash/data" "$trash/etc/config"
+    echo_n_run mv "$datadir" "$trash/data"
+    if [[ "$system_wide" == true ]]; then
+      echo_n_run mv "$globalconfigdir" "$trash/etc/config"
     fi
   fi
 }
 
+# Copy the default config file to a directory and set it up
+# Args:
+# $1: the path to a directory where the config file will be copied to
+function copy_config_file_to()
+{
+  local -r to="$1"
+  local -r config_file_name="kworkflow.config"
+  local -r config_file_template="etc/kworkflow_template.config"
+
+  mkdir -p "$to"
+  echo_n_run cp "$config_file_template" "$to/$config_file_name"
+  sed -i -e "s/USERKW/$USER/g" -e "s,KW_SOUNDS_DIR,$sharedir/sounds/,g" \
+         -e "/^#?.*/d" "$to/$config_file_name"
+}
+
+# Set up both user-global and system-global config files
 function setup_config_file()
 {
-  say "Setting up global configuration file"
-  local config_files_path="$INSTALLTO/$CONFIG_DIR"
-  local config_file_template="$config_files_path/kworkflow_template.config"
-  local global_config_name="kworkflow.config"
+  local -r config_file_template="etc/kworkflow_template.config"
+
+  say "Setting up configuration files"
 
   if [[ -f "$config_file_template" ]]; then
-    cp "$config_file_template" "$config_files_path/$global_config_name"
-    sed -i -e "s/USERKW/$USER/g" -e "s,INSTALLPATH,$INSTALLTO,g" \
-           -e "/^#?.*/d" "$config_files_path/$global_config_name"
+    copy_config_file_to "$data"
+    if [[ "$system_wide" == true ]]; then
+      copy_config_file_to "$globalconfigdir"
+    fi
   else
     warning "setup could not find $config_file_template"
   fi
 }
 
-function synchronize_fish()
+function synchronize_completion_files()
 {
-    local kw_fish_path="set -gx PATH $PATH:$HOME/.config/kw"
+  # TODO: allow completion for local installation
```

Comment:
If we handle the local installation as
```
# Note that I added ".bin", ".lib"...
if [[ "$HOME" -ef "$prefix" ]]; then
declare -r binpath="$prefix/.bin/kw"
declare -r libdir="$prefix/.lib/kw/"
declare -r sharedir="$prefix/.share/kw/"
...
fi
```
we can fix the auto-completion issue by keep it with `src` or loading it in the `kw` file

Reply:
Hmm, I don't like keeping the autocomplete scripts within `src`. I feel like it gets a little messy. But even if we did keep it there and made `kw` load it, it still wouldn't work because the completion script would only get loaded after `kw` is executed. And we use `kw<tab><tab>` before even invoking `kw`....

The current solution we have is to source the script in the user's bashrc file. We can still use this solution, but I personally think it's kind of unpleasant to edit the user's bashrc.

Reply:
hmm... I don't think that add something into the user `bashrc` is a huge issue, I'm not sure but I think that `rvm` and `pip` uses a similar approach. IMHO, if `setup.sh` correctly handles the addition and removal of the extra line everything is fine.